iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -98,659 +31,752 @@ function localName(e) {
-    var height = $("html",this._contents).height();
-    $('body', this._contents).css('zoom',this.scale);
+    zoomIn() {
+        this.scale *= 1.1;
+        this._zoom();
+    }
 
-    var newHeight = $("html", this._contents).height();
-    this._contents.scrollTop(newHeight * (viewTop)/height );
-}
+    zoomOut() {
+        this.scale /= 1.1;
+        this._zoom();
+    }
 
-Viewer.prototype.zoomIn = function () {
-    this.scale *= 1.1;
-    this._zoom();
-}
+    factsInSameTable(fact) {
+        var facts = [];
+        const e = this.elementsForItemId(fact.id);
+        e.closest("table").find(".ixbrl-element").each(function () {
+            facts = facts.concat($(this).data('ivid'));
+        });
+        return facts;
+    }
 
-Viewer.prototype.zoomOut = function () {
-    this.scale /= 1.1;
-    this._zoom();
-}
+    linkedHighlightFact(f) {
+        this.changeItemClass(f.id, "ixbrl-linked-highlight");
+    }
 
-Viewer.prototype.factsInSameTable = function (fact) {
-    var facts = [];
-    const e = this.elementsForItemId(fact.id);
-    e.closest("table").find(".ixbrl-element").each(function () {
-        facts = facts.concat($(this).data('ivid'));
-    });
-    return facts;
-}
+    clearLinkedHighlightFact(f) {
+        this.changeItemClass(f.id, "ixbrl-linked-highlight", true);
+    }
 
-Viewer.prototype.linkedHighlightFact = function (f) {
-    this.changeItemClass(f.id, "ixbrl-linked-highlight");
-}
+    _setTitle(docIndex) {
+        $('#top-bar .document-title').text($('head title', this._iframes.eq(docIndex).contents()).text());
+    }
 
-Viewer.prototype.clearLinkedHighlightFact = function (f) {
-    this.changeItemClass(f.id, "ixbrl-linked-highlight", true);
-}
+    showDocumentForItemId = function(itemId) {
+        this.selectDocument(this._ixNodeMap[itemId].docIndex);
+    }
 
-Viewer.prototype._setTitle = function (docIndex) {
-    $('#top-bar .document-title').text($('head title', this._iframes.eq(docIndex).contents()).text());
-}
+    currentDocument() {
+        return this._iframes.eq(this._currentDocumentIndex);
+    }
 
-Viewer.prototype.showDocumentForItemId = function(itemId) {
-    this.selectDocument(this._ixNodeMap[itemId].docIndex);
-}
+    selectDocument(docIndex) {
+        this._currentDocumentIndex = docIndex;
+        $('#ixv #viewer-pane .ixds-tabs .tab')
+            .removeClass("active")
+            .eq(docIndex)
+            .addClass("active");
+        /* Show/hide documents using height rather than display property to avoid a
+         * delay when switching tabs on large, slow-to-render documents. */
+        this._iframes
+            .height(0)
+            .data("selected", false)
+            .eq(docIndex)
+            .height("100%")
+            .data("selected", true);
+        this._setTitle(docIndex);
+    }
 
-Viewer.prototype.currentDocument = function () {
-    return this._iframes.eq(this._currentDocumentIndex);
-}
+    * postProcess(iframe) {
```

Comment:
`iframe` isn't passed in or used currently.

If `iframe` is removed, then the `*` can be removed as well and line 779 could be updated to:
```
runGenerator(this.postProcess);
```

Reply:
Done in 03f2df9

Reply:
I've removed the unused `iframe` parameter, but I get a parse error if I attempt to remove the `*` because it uses `yield`.

`==` replaced with `===` in 03f2df98726dcdeef4f56fc0574a526c7e5efe8c